service/src/lib.rs

```diff
@@ -142,6 +151,10 @@ impl PolkadotService for Service<LightComponents<Factory>> {
 		Service::client(self)
 	}
 
+	fn select_chain(&self) -> client::LongestChain<Self::Backend, Block> {
+		unimplemented!();
```

Comment:
still this to do

Reply:
either I implement a new trait FullPolkadotService, but then Worker trait will need to use this new one, it is not correct no ?
or I can make it a runtime check with an option

Reply:
cc @gnunicorn : how should we do this?

Reply:
Well, what is `&self` here? Does it know of `backend`? then it could instantiate its own `LongestChain` (as this is already hard-referenced anyways).

However, there [was a follow up PR merged yesterday](https://github.com/paritytech/substrate/pull/2574), which addresses the light-client problem by making it an `Option<SelectChain>` and then this would become `None`. Maybe we want to catch up just a slight bit further before merging?

Reply:
done